src/sentry/static/sentry/app/views/settings/organizationRelay/list/cardHeader.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {t, tct} from 'app/locale';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import QuestionTooltip from 'app/components/questionTooltip';
+import DateTime from 'app/components/dateTime';
+import {IconEdit, IconDelete} from 'app/icons';
+import space from 'app/styles/space';
+import {Relay} from 'app/types';
+import Clipboard from 'app/components/clipboard';
+
+type Props = Relay & {
+  onEditKey: (publicKey: Relay['publicKey']) => () => void;
+  onDeleteKey: (publicKey: Relay['publicKey']) => () => void;
+};
+
+const CardHeader = ({
+  publicKey,
+  name,
+  description,
+  created,
+  onEditKey,
+  onDeleteKey,
+}: Props) => (
+  <Header>
+    <MainInfo>
+      <Name>
+        <div>{name}</div>
+        {description && <QuestionTooltip position="top" size="sm" title={description} />}
+      </Name>
+      <Date>
+        {tct('Created on [date]', {date: <DateTime date={created} timeAndDate />})}
+      </Date>
+    </MainInfo>
+    <ButtonBar gap={1}>
+      <Clipboard value={publicKey}>
+        <Button size="small">{t('Copy Key')}</Button>
+      </Clipboard>
+      <Button
+        size="small"
+        onClick={onEditKey(publicKey)}
+        icon={<IconEdit />}
+        label={t('Edit Key')}
+      />
+      <Button
+        size="small"
+        onClick={onDeleteKey(publicKey)}
+        icon={<IconDelete />}
+        label={t('Delete Key')}
+      />
```

Comment:
Can we please wrap this in a Confirm?

Reply:
sure thing ..thanks for the suggestion 😉